ocrd-olena-binarize

```diff
@@ -6,52 +6,60 @@ which ocrd >/dev/null 2>/dev/null || { echo "ocrd not in \$PATH. Panicking"; exi
 
 SHAREDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 
-declare -A mimetype2extension=(
-    ['image/tiff']='.tiff'
-    ['image/jp2']='.jp2'
-    ['image/png']='.png'
-)
-
 main () {
     source `ocrd bashlib filename`
     ocrd__wrap "$SHAREDIR/ocrd-tool.json" "ocrd-olena-binarize" "$@"
 
     cd "${ocrd__argv[working_dir]}"
+    in_file_grp=${ocrd__argv[input_file_grp]}
+    out_file_grp=${ocrd__argv[output_file_grp]}
+    mkdir -p $out_file_grp
 
-    ocrd workspace find              \
-        -G "${ocrd__argv[input_file_grp]}" \
+    local IFS=$'\n'
+    files=($(ocrd workspace find     \
+        -G $in_file_grp              \
         -k local_filename            \
         -k ID                        \
         -k mimetype                  \
-        --download                   \
-    | while read -r csv;do
+        -k pageId                    \
+        --download))
+    for csv in "${files[@]}"; do
         # Parse comma separated fields
         local IFS=$'\t'
```

Comment:
What is `ISF`? It is defined twice but never accessed.

Reply:
`IFS` is bash's _internal field separator_ variable. It is needed here to mark the boundaries of array elements (newline vs tab). First, we only want the newlines when we init the `files` array. Then inside one line we only want tabs to init the `fields` array.